src/entry-point-client/App.tsx

```diff
@@ -1,21 +1,57 @@
 import { hot } from "react-hot-loader";
-import { Route, Switch } from "react-router-dom";
+import { Redirect, Route, Switch } from "react-router-dom";
 
 import { GlobalStyles } from "../styles";
 import PageList from "./PageList";
 import WindowTitle from "./WindowTitle";
+import PrivateRoute from "../private-route/PrivateRoute";
+import Loading from "../loading-spinner/Loading";
+import VerificationNeeded from "../verification-needed/VerificationNeeded";
+
+import { useAuth0 } from "../react-auth0-spa";
 
 const App: React.FC = () => {
+  const { loading, user } = useAuth0();
+
+  if (loading) {
+    return <Loading />;
+  }
+
+  const redirectIfUnverified = (userObject, contents) => {
+    if (userObject && !userObject.email_verified) {
+      return <Redirect to="/verify" />
+    }
+    return contents;
+  }
+
   return (
     <>
       <GlobalStyles />
       <Switch>
-        {PageList.map(({ path, title, contents }) => (
-          <Route key={path} path={path} exact>
-            <WindowTitle>{title}</WindowTitle>
-            {contents}
-          </Route>
-        ))}
+        <Route path="/verify">
```

Comment:
This is listed manually and first instead of being listed in `PageList` to ensure that it will be defined ahead of time in the event that a given route below gets converted into a Redirect to /verify.